src/lib/instruction-parser.ts

```diff
@@ -119,6 +128,109 @@ export class InstructionParser {
     }
   }
 
+  async getParsedEvents(tx: TransactionWithMeta, connection: Connection) {
+    const events: ParsedEvent[] = [];
+    const routingInstructionIndexes: number[] = [];
+    tx.transaction.message.instructions.forEach((instruction, index) => {
+      if (instruction.programId.equals(this.programId)) {
+        const ix = this.coder.instruction.decode(
+          (instruction as PartialInstruction).data,
+          "base58"
+        );
+        if (this.isRouting(ix.name)) routingInstructionIndexes.push(index);
+      }
+    });
+
+    const innerInstructions: (PartialInstruction | ParsedInstruction)[] = [];
+    const swapInstructionIndexes = [];
+
+    for (const routingInstructionIndex of routingInstructionIndexes) {
+      for (const instructions of tx.meta.innerInstructions) {
+        if (instructions.index === routingInstructionIndex) {
+          instructions.instructions.forEach((instruction, index) => {
+            innerInstructions.push(instruction);
+            if (instruction.programId.toBase58() in AMM_TYPES) {
+              swapInstructionIndexes.push(index);
+            }
+          });
+          break;
+        }
+      }
+    }
+
+    for (const index of swapInstructionIndexes) {
+      const swapInstruction = innerInstructions[index];
+      const [inTransferInstruction, outTransferInstruction] =
```

Comment:
i think it is easier to figure out the input and output token accounts from the respective AMM and figure out the actual amount that is transfer in and out. sometimes, some AMMs do it twice.

Reply:
As there is no common way to figure out input and output tokens for all AMMs. I think there is a need to implement custom parsing logic to parse token transfers for few AMMs to address such cases. There are few other cases too where AMMs require custom parsing. Please correct me if I missed something on this